src/modules/launcher/PowerLauncher/Plugin/QueryBuilder.cs

```diff
@@ -29,11 +30,28 @@ public static class QueryBuilder
                     continue;
                 }
 
+                // Save the length of the longest matching keyword for later
+                if (pluginActionKeyword.Length > longestActionKeywordLength)
+                {
+                    longestActionKeywordLength = pluginActionKeyword.Length;
+                }
+
                 // A new query is constructed for each plugin
                 var query = new Query(text, pluginActionKeyword);
                 pluginQueryPair.TryAdd(plugin, query);
             }
 
+            // If we have plugin action keywords that start with the same char we get false positives (Example: ? and ??)
+            // Here we remove each query pair that has a shorter keyword than the longest matching one
+            var pairList = pluginQueryPair;
```

Comment:
We need this here because we can't change collections/vars that we are iterating at the moment of changing them. Maybe we have to call a `.copy()` method. But I am not sure at the moment.

Reply:
why not? afaik we can if we're using `foreach`

Reply:
Is this only on for loops? I thought I read that somewhere.

Reply:
Yeah that's only if your accessing by index.

Either doing it like this is better: `foreach (var plugin in pluginQueryPair.Keys)`

Reply:
Interesting. Here I need the `.ToList()` to be able to change the original collection: https://github.com/microsoft/PowerToys/blob/main/src/modules/launcher/PowerLauncher/Helper/EnvironmentHelper.cs#L94

But this is a better solution and we should use it here too. 😉

Reply:
>Here I need the `.ToList()` to be able to change the original collection

You don't actually, you're not mutating `varName`.
`.ToList()` return a new object

Reply:
> >Here I need the `.ToList()` to be able to change the original collection
> 
> You don't actually, you're not mutating `varName`.

If we switch to iterate `varName.Keys` directly then we need it. Because then we can't change `varName` in the foreach because it's looked.

> `.ToList()` return a new object btw.

I know that.

Reply:
@cyberrex5 Sory im was wrong. You are right. It's working without `.ToList()`.